src/libmeasurement_kit/ooni/ooni_test.cpp

```diff
@@ -77,30 +78,27 @@ void OoniTest::geoip_lookup(Callback<> cb) {
                     logger->debug("saving user's real ip on user's request");
                     probe_ip = ip;
                 }
-                std::string country_p =
-                    options.get("geoip_country_path", std::string{});
-                std::string asn_p =
-                    options.get("geoip_asn_path", std::string{});
-                if (country_p == "" or asn_p == "") {
+                auto cntry_p = options.get("geoip_country_path", std::string{});
+                auto asn_p = options.get("geoip_asn_path", std::string{});
+                if (cntry_p == "" or asn_p == "") {
                     logger->warn("geoip files not configured; skipping");
                 } else {
-                    ErrorOr<nlohmann::json> res = geoip(ip, country_p, asn_p);
-                    if (!!res) {
-                        logger->debug("GeoIP result: %s", res->dump().c_str());
-                        // Since `geoip()` sets defaults before querying, the
-                        // following accesses of json should not fail unless for
-                        // programmer error after refactoring. In that case,
-                        // better to let the exception unwind than just print
-                        // a warning, because the former is easier to notice
-                        // and therefore fix during development
+                    auto gi = IpLocationProxy::global()->open(
+                                cntry_p, asn_p, "", logger);
+                    ErrorOr<std::string> s;
+                    s = gi->resolve_asn(ip);
+                    if (!!s) {
+                        logger->info("probe_asn: %s", s->c_str());
                         if (options.get("save_real_probe_asn", true)) {
```

Comment:
Since we actually don't do anything with the lookup when it fails, we may want to just skip doing it if `save_real_probe_XXX` is false.